eng/install-test-runner-prerequisites.ps1

```diff
@@ -0,0 +1,71 @@
+﻿# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+# Run this script as an administrator to install Chocolatey, Pyenv, Python 3.9.4, and Poetry.
+# This script should be run should be run once for environments that do not
+# already have these prerequisites set up.
+
+function Update-Path {
+    $env:Path = [System.Environment]::GetEnvironmentVariable("Path","Machine") +
+                ";" +
+                [System.Environment]::GetEnvironmentVariable("Path","User")
+}
+
+function Install-Chocolatey {
+    if (! (Get-Command choco.exe -ErrorAction SilentlyContinue )) {
+        Set-ExecutionPolicy Bypass -Scope Process -Force
+        Invoke-WebRequest https://chocolatey.org/install.ps1 -UseBasicParsing | Invoke-Expression
+        $env:ChocolateyInstall = Convert-Path "$((Get-Command choco).path)\..\.."
+        Import-Module "$env:ChocolateyInstall\helpers\chocolateyProfile.psm1"
+        refreshenv
+    }
+
+}
+
+function Install-Pyenv {
+    $pyenvVersion = cmd /c pyenv --version
+    if(!$pyenvVersion -or $pyenvVersion -notlike 'pyenv 2.*'){
+        choco install pyenv-win -y
+        refreshenv
+        # refreshenv doesn't appear to be sufficient to recognize user environment variable changes
+        Update-Path
+    }
+}
+
+function Install-Python {
+    pyenv install 3.9.4
+    pyenv rehash
+    pyenv local 3.9.4
+}
+
+function Install-Poetry {
+    # Ensure pip is on the latest version
+    python -m pip install --upgrade pip
+
+    # Install poetry
+    # Poetry's native installation process encounters SSL errors
+    # in some environments. `pip install` is a reasonable alternative
+    # that has been shown to work in our situation.
+    pip install --user poetry
```

Comment:
I really dislike having to do that, but I spent hours trying to find a work around to no avail.